### PR TITLE
feat: support identifiers, first_screen, direct_sign_in and extra_params sign-in params

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Add following sign-in params support:
1. `first_screen`: This parameter allows you to customize the first screen that users see when they start the authentication process, see [reference](https://docs.logto.io/docs/references/openid-connect/authentication-parameters/#direct-sign-in).
2. `identifiers`: should work with `first_screen`
3. `direct_sign_in`: This parameter allows you to skip the first screen and invoke the sign-in process directly, see [reference](https://docs.logto.io/docs/references/openid-connect/authentication-parameters/#direct-sign-in).
4. `extra_params`: provides the ability of taking any other extra params

## Testing

Tested locally.

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: samples/sample-blazor/Program.cs [hunk 1]
```diff
@@ -47,8 +47,37 @@
...
+            RedirectUri = "/" 
+        };
+
+        authProperties.SetParameter("first_screen", LogtoParameters.Authentication.FirstScreen.Register);
+        authProperties.SetParameter("identifiers", string.Join(",", new[] 
+        { 
+            LogtoParameters.Authentication.Identifiers.Username,
+        }));
+
+        // Set `direct_sign_in`
```
Comment (gao-sun) [inline samples/sample-blazor/Program.cs:61]: this is a stop sign comment. briefly introduce the feature and link to a meaningful url, or just remove it.

File: samples/sample-blazor/Program.cs [hunk 2]
```diff
@@ -47,8 +47,37 @@
...
+        var directSignIn = new LogtoParameters.Authentication.DirectSignIn
+        {
+            Target = "github",
+            Method = LogtoParameters.Authentication.DirectSignIn.Methods.Social
+        };
+        authProperties.SetParameter("direct_sign_in", System.Text.Json.JsonSerializer.Serialize(directSignIn));
+
+        // Set `extra_params`
```
Comment (gao-sun) [inline samples/sample-blazor/Program.cs:69]: this is a stop sign comment. briefly introduce the feature and link to a meaningful url, or just remove it.

File: samples/sample-blazor/Program.cs [hunk 3]
```diff
@@ -47,8 +47,37 @@
...
+        var extraParams = new LogtoParameters.Authentication.ExtraParams
+        {
+            { "utm_source", "website" },
+            { "utm_medium", "organic" }
```
Comment (gao-sun) [inline samples/sample-blazor/Program.cs:73]: `utm_*` means nothing in Logto. why?

File: samples/sample-mvc/Controllers/HomeController.cs [hunk 4]
```diff
@@ -25,7 +26,32 @@ public async Task<IActionResult> Index()
 
     public IActionResult SignIn()
     {
-        return Challenge(new AuthenticationProperties { RedirectUri = "/" });
+        var authProperties = new AuthenticationProperties
```
Comment (gao-sun) [inline samples/sample-mvc/Controllers/HomeController.cs:29]: ditto

File: samples/sample/Pages/Index.cshtml.cs [hunk 5]
```diff
@@ -22,7 +23,32 @@ public async Task OnGetAsync()
 
     public async Task OnPostSignInAsync()
     {
-        await HttpContext.ChallengeAsync(new AuthenticationProperties { RedirectUri = "/" });
+        var authProperties = new AuthenticationProperties
```
Comment (gao-sun) [inline samples/sample/Pages/Index.cshtml.cs:26]: ditto

File: src/Logto.AspNetCore.Authentication/LogtoParameters.cs [hunk 6]
```diff
@@ -115,4 +116,93 @@ public static class Claims
...
+      public const string SingleSignOn = "single_sign_on";
+
+      /// <summary>
+      /// Show the reset password form first.
+      /// </summary>
+      public const string ResetPassword = "reset_password";
+    }
+
+    /// <summary>
+    /// The identifiers to use for authentication.
```
Comment (gao-sun) [inline src/Logto.AspNetCore.Authentication/LogtoParameters.cs:152]: should mention it's for first screen only?

Comment (darcyYe) [reply src/Logto.AspNetCore.Authentication/LogtoParameters.cs:152]: Updated.

File: samples/sample-blazor/Program.cs (hunk 3 above)
Comment (darcyYe) [reply samples/sample-blazor/Program.cs]: The original intention was to demonstrate how to use `extra_params` through examples. To align with other SDKs, removed this `extra_params` example part.

**Pull request merged**